packages/core/src/RenderingEngine/helpers/setDefaultVolumeVOI.ts

```diff
@@ -0,0 +1,234 @@
+  }
+
+  voi = handlePreScaledVolume(imageVolume, voi);
+  const { lower, upper } = voi;
+
+  volumeActor
+    .getProperty()
+    .getRGBTransferFunction(0)
+    .setMappingRange(lower, upper);
+}
+
+function handlePreScaledVolume(imageVolume: IImageVolume, voi: VOIRange) {
+  const imageIds = imageVolume.imageIds;
+  const imageIdIndex = Math.floor(imageIds.length / 2);
+  const imageId = imageIds[imageIdIndex];
+
+  const generalSeriesModule =
+    metaData.get('generalSeriesModule', imageId) || {};
+
+  /**
+   * If the volume is prescaled and the modality is PT Sometimes you get super high
+   * values at the peak and it skews the min/max so nothing useful is displayed
+   * Therefore, we follow the majority of other viewers and we set the min/max
+   * for the scaled PT to be 0, 5
+   */
+  if (generalSeriesModule.modality === 'PT' && imageVolume.isPrescaled) {
+    return {
+      lower: 0,
+      upper: 5,
+    };
+  }
+
+  return voi;
+}
+
+/**
+ * Get the VOI from the metadata of the middle slice of the image volume. It checks
+ * the metadata for the VOI and if it is not found, it returns null
+ *
+ * @param imageVolume - The image volume that we want to get the VOI from.
+ * @returns VOIRange with lower and upper values
+ */
+function getVOIFromMetadata(imageVolume: IImageVolume): VOIRange {
+  const { imageIds } = imageVolume;
+
+  const imageIdIndex = Math.floor(imageIds.length / 2);
+  const imageId = imageIds[imageIdIndex];
+
+  const voiLutModule = metaData.get('voiLutModule', imageId);
+
+  if (voiLutModule && voiLutModule.windowWidth && voiLutModule.windowCenter) {
+    const { windowWidth, windowCenter } = voiLutModule;
+
+    const voi = {
+      windowWidth: Array.isArray(windowWidth) ? windowWidth[0] : windowWidth,
+      windowCenter: Array.isArray(windowCenter)
+        ? windowCenter[0]
+        : windowCenter,
+    };
+
+    const { lower, upper } = windowLevel.toLowHighRange(
+      Number(voi.windowWidth),
+      Number(voi.windowCenter)
+    );
+
+    return {
+      lower,
+      upper,
+    };
+  }
+}
+
+/**
+ * It loads the middle slice image (middle imageId) and based on its min
+ * and max pixel values, it calculates the VOI.
+ *
+ * @param imageVolume - The image volume that we want to get the VOI from.
+ * @returns The VOIRange with lower and upper values
+ */
+async function getVOIFromMinMax(imageVolume: IImageVolume): Promise<VOIRange> {
+  const { scalarData, imageIds } = imageVolume;
+
+  // Get the middle image from the list of imageIds
+  const imageIdIndex = Math.floor(imageIds.length / 2);
+  const imageId = imageVolume.imageIds[imageIdIndex];
+  const generalSeriesModule =
+    metaData.get('generalSeriesModule', imageId) || {};
+  const { modality } = generalSeriesModule;
+  const modalityLutModule = metaData.get('modalityLutModule', imageId) || {};
+
+  const numImages = imageIds.length;
+  const bytesPerImage = scalarData.byteLength / numImages;
+  const length = scalarData.length / numImages;
```

Comment:
Can we call this variable `voxelsPerImage` or something? Think its a bit cleaner than a generic `length`.